r55/src/error.rs

```diff
@@ -0,0 +1,58 @@
+//! R55 crate errors
+
+use revm::{
+    primitives::{EVMError, ExecutionResult},
+    Database, InMemoryDB,
+};
+use rvemu::exception::Exception;
+
+pub type Result<T> = core::result::Result<T, Error>;
+
+/// Error encountered on RISC-V execution
+#[allow(clippy::enum_variant_names)]
+#[derive(Debug, thiserror::Error)]
+pub enum Error<DB: Database = InMemoryDB>
+where
+    DB::Error: std::error::Error + 'static,
+{
+    /// The exception kind on RISC-V [`emulator`](`Emulator`)
+    #[error("Got RISC-V emulator exception: {0:?}")]
+    RvEmuException(Exception),
+    /// EVM error
+    #[error(transparent)]
+    EvmError(#[from] EVMError<DB::Error>),
```

Comment:
where do we use this one?

Reply:
on evm transact we use EVM error, exception is on getting dram slice